### PR TITLE
Issue/2383

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Try to fix http://bugs.vcmi.eu/view.php?id=2383

### CODE REVIEW

**Comment by ArseniyShestakov** [comment]
I'm not expert in networking part of code, but I feel that naming of all these variables/conditions is a bit wrong and sendPackageResponse just tells client that server had applier for certain package.

So I suspect your code may end up VCAI stop attempt to move hero, but that is not a fix for actual bugs that caused loop in first place.

File: AI/VCAI/VCAI.cpp [hunk 1]
```diff
@@ -2002,6 +2002,8 @@ bool VCAI::moveHeroToTile(int3 dst, HeroPtr h)
 		completeGoal (sptr(Goals::VisitTile(dst).sethero(h))); //we stepped on some tile, anyway
 		completeGoal (sptr(Goals::ClearWayTo(dst).sethero(h)));
 
+		ret = dst == h->visitablePos();
```
Comment (alexvins) [inline AI/VCAI/VCAI.cpp]: There is very suspicious code above.
```cpp

for(; i>0; i--)
{
//...
}   

ret = !i; 
```
"i" can`t be 0.

**Comment by ArseniyShestakov** [comment]
Not saying that code is valid, but AI shouldn't start to move in first place if it's don't have sufficient amount of movement points.

**Comment by DjWarmonger** [comment]
In this case the sufficient amount of point is 0, as far as I understand. AI should be always able to revisit its current location (eg. dwelling on day 1).

**Comment by ArseniyShestakov** [comment]
As far as I see it's not movement on same tile, but normal diagonal movement with hero having something like 121 movement points. Reason it's fail is because cost calculation is inconsistent between pathfinder on client and server.

**Comment by vmarkovtsev** [comment]
Added () for clarity and removed the useless ``ret`` assignment.

**Comment by vmarkovtsev** [comment]
Lets's merge this symptom suppression (it's still reasonable) and continue with another PR for the root cause.

**Comment by ArseniyShestakov** [comment]
Agree. Don't worry about pathfinding / movement bug. I'm working on it.

**Pull request merged**